cmd/puma-dev/main_test.go

```diff
@@ -294,3 +191,115 @@ func TestMain_allCheck_badArg(t *testing.T) {
 	assert.True(t, ok)
 	assert.False(t, exit.Success())
 }
+
+func generateLivePumaDevCertIfNotExist(t *testing.T) {
+	liveSupportPath := homedir.MustExpand(dev.SupportDir)
+	liveCertPath := filepath.Join(liveSupportPath, "cert.pem")
+	liveKeyPath := filepath.Join(liveSupportPath, "key.pem")
+
+	if !FileExists(liveCertPath) || !FileExists(liveKeyPath) {
+		MakeDirectoryOrFail(t, liveSupportPath)
+
+		if err := dev.GeneratePumaDevCertificateAuthority(liveCertPath, liveKeyPath); err != nil {
+			assert.FailNow(t, err.Error())
+		}
+	}
+}
+
+func launchPumaDevBackgroundServerWithDefaults(t *testing.T) func() {
+	address := fmt.Sprintf("localhost:%d", *fHTTPPort)
+	timeout := time.Duration(10 * time.Second)
+
+	if _, err := net.DialTimeout("tcp", address, timeout); err == nil {
+		return func() {}
+	}
+
+	generateLivePumaDevCertIfNotExist(t)
+
+	StubCommandLineArgs()
+	SetFlagOrFail(t, "dir", testAppLinkDirPath)
+	MakeDirectoryOrFail(t, testAppLinkDirPath)
+
+	testAppsToLink := map[string]string{
+		"hipuma":      "rack-hi-puma",
+		"static-site": "static-hi-puma",
+	}
```

Comment:
feels kinda bad burying these all the way down here, but they're not used outside this function, so shrug.